src/modules/esl-carousel/plugin/touch/esl-carousel.touch.mixin.ts

```diff
@@ -107,13 +113,22 @@ export class ESLCarouselTouchMixin extends ESLCarouselPlugin {
     const offset = this.getOffset(event);
 
     if (!this.$host.hasAttribute('dragging')) {
+      // Prevents draggable state if the content is scrolled
+      if (isOffsetChanged(this.startEventOffset)) return this._onPointerUp(event);
+      // Prevents draggable state if the text is selected
+      if (document.getSelection()?.isCollapsed === false) return this._onPointerUp(event);
+      // Early exit, if the swipe timeout tolerance is not reached
+      if (this.isSwipeMode || event.timeStamp - this.startTimestamp > this.swipeTimeout) return this._onPointerUp(event);
```

Comment:
Maybe it makes sense to combine this condition with the condition above

Reply:
Did you mean to group conditions here and in the method above?

Because the combining of exact this group of conditions commented in the separate ifs will significantly reduce readability of already quite complex logic

Reply:
@ala-n thank, now I realize comments are very important here.